v3ioutils/container.go

```diff
@@ -198,19 +201,22 @@ func getItemsWorker(container v3io.Container, input *v3io.GetItemsInput, fileNam
 	}
 }
 
-func deleteObjectWorker(tablePath string, container v3io.Container, fileNameChan <-chan string, terminationChan chan<- error, onErrorTerminationChannel <-chan struct{}) {
+func deleteObjectWorker(tablePath string, container v3io.Container, fileNameChan <-chan string, terminationChan chan<- error,
+	onErrorTerminationChannel <-chan struct{}, deleteSchema bool) {
 	for {
 		select {
 		case fileName, ok := <-fileNameChan:
 			if !ok {
 				terminationChan <- nil
 				return
 			}
-			input := &v3io.DeleteObjectInput{Path: tablePath + "/" + fileName}
-			err := container.DeleteObjectSync(input)
-			if err != nil {
-				terminationChan <- err
-				return
+			if deleteSchema || (!deleteSchema && fileName != ".#schema") {
```

Comment:
redundant check for the second `!deleteSchema`. you can just write:
`if deleteSchema || fileName != ".#schema"`

Reply:
Or maybe better, to filter out the schema file in `getItemsWorker` and here to delete everything that is returned without any conditions (might be a bit faster 🤷‍♂ ).